app/uk/gov/hmrc/cipphonenumberverification/services/VerifyHelper.scala

```diff
@@ -104,50 +111,83 @@ abstract class VerifyHelper @Inject()(otpService: OtpService, auditService: Audi
     maybePhoneNumberAndOtp match {
-      case Some(storedPhoneNumberAndOtp) => checkIfPasscodeMatches(enteredPhoneNumberAndOtp, storedPhoneNumberAndOtp)
+      case Some(storedPhoneNumberAndOtp) => processFoundPhoneNumberPasscodeData(enteredPhoneNumberAndOtp, storedPhoneNumberAndOtp)
       case _ => auditService.sendExplicitAuditEvent(PhoneNumberVerificationCheck,
-        VerificationCheckAuditEvent(enteredPhoneNumberAndOtp.phoneNumber, enteredPhoneNumberAndOtp.otp, "Not verified"))
-        Future.successful(Ok(Json.toJson(VerificationStatus("Not verified"))))
+        VerificationCheckAuditEvent(enteredPhoneNumberAndOtp.phoneNumber, enteredPhoneNumberAndOtp.otp, NOT_VERIFIED))
+        Future.successful(Ok(Json.toJson(VerificationStatus(NOT_VERIFIED))))
     }
 
+  private def processFoundPhoneNumberPasscodeData(enteredPhoneNumberAndOtp: PhoneNumberAndOtp, foundPhoneNumberPasscodeData: PhoneNumberPasscodeData)(implicit hc: HeaderCarrier): Future[Result] = {
+    val now = dateTimeUtils.getCurrentDateTime()
+    hasPasscodeAllowedToBeStoredTimeElapsed(foundPhoneNumberPasscodeData: PhoneNumberPasscodeData, now) match {
+      case true =>
+        auditService.sendExplicitAuditEvent(PhoneNumberVerificationCheck,
+          VerificationCheckAuditEvent(enteredPhoneNumberAndOtp.phoneNumber, enteredPhoneNumberAndOtp.otp, NOT_VERIFIED))
+        Future.successful(Ok(Json.toJson(ErrorResponse(VERIFICATION_ERROR, PASSCODE_STORED_TIME_ELAPSED))))
+      case false => checkIfPasscodeIsStillAllowedToBeUsed(enteredPhoneNumberAndOtp, foundPhoneNumberPasscodeData, now)
+    }
+  }
+
+  private def checkIfPasscodeIsStillAllowedToBeUsed(enteredPhoneNumberAndOtp: PhoneNumberAndOtp, foundPhoneNumberPasscodeData: PhoneNumberPasscodeData, now: LocalDateTime)(implicit hc: HeaderCarrier): Future[Result] = {
+    hasPasscodeAllowedToBeUsedTimeElapsed(foundPhoneNumberPasscodeData: PhoneNumberPasscodeData, now) match {
+      case true =>
+        auditService.sendExplicitAuditEvent(PhoneNumberVerificationCheck,
+          VerificationCheckAuditEvent(enteredPhoneNumberAndOtp.phoneNumber, enteredPhoneNumberAndOtp.otp, NOT_VERIFIED))
+        Future.successful(Ok(Json.toJson(ErrorResponse(VERIFICATION_ERROR, PASSCODE_ALLOWED_TIME_ELAPSED))))
+      case false => checkIfPasscodeMatches(enteredPhoneNumberAndOtp, foundPhoneNumberPasscodeData)
+    }
+  }
+
+  private def hasPasscodeAllowedToBeUsedTimeElapsed(foundPhoneNumberPasscodeData: PhoneNumberPasscodeData, currentTime: LocalDateTime): Boolean = {
+    val elapsedTimeInMilliseconds: Long = calculateElapsedTime(foundPhoneNumberPasscodeData.createdAt, currentTime)
+    val allowedTimeGapForPasscodeUsageInMilliseconds: Long = Duration.ofMinutes(config.passcodeExpiry).toMillis
+    elapsedTimeInMilliseconds > allowedTimeGapForPasscodeUsageInMilliseconds
+  }
+
+  private def hasPasscodeAllowedToBeStoredTimeElapsed(foundPhoneNumberPasscodeData: PhoneNumberPasscodeData, currentTime: LocalDateTime): Boolean = {
```

Comment:
is this not identical to `hasPasscodeAllowedToBeUsedTimeElapsed`?

Reply:
No its not identical, its the equivalent for the has the cache expired.  (2 hours)

Reply:
as above, I have refactored them to be hasPasscodeExpired and hasCacheExpired

Reply:
I think we have discussed this a few times and we said it is safer to use a createdOn timestamp.  
Since doing this ticket I have also discovered the following:

- the metadata timestamps in MongoDB are in UTC.  (an hour out at the moment). which is fine but our times in the application would also have to be in UTC, which could be done.  probably a good thing as its future proofing our application and would make it work globally!  (i.e. timezone independent).  you never know!

- The metadata in MongoDB is in MongoDB and our logic will always be in the application and therefore a timestamp would still have to be added to the data object.  Therefore its back to setting the timestamp in the application.  (as i don't know if there is a way to populate a data object with a metadata field in MongoDB if you see what I mean).

What are your thoughts on the above?

Reply:
I will update the code to use UTC and refactor the cache expiry code as discussed

Reply:
updated